powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/PowerAuthFido2Client.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2023 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package com.wultra.security.powerauth.client;
+
+import com.wultra.security.powerauth.client.model.entity.fido2.AuthenticatorAssertionResponse;
+import com.wultra.security.powerauth.client.model.entity.fido2.AuthenticatorParameters;
+import com.wultra.security.powerauth.client.model.error.PowerAuthClientException;
+import com.wultra.security.powerauth.client.model.request.fido2.*;
+import com.wultra.security.powerauth.client.model.response.fido2.*;
+import org.springframework.util.MultiValueMap;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * PowerAuth FIDO2 client interface.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+public interface PowerAuthFido2Client {
+
+    /**
+     * Get list of registered authenticators for a user.
+     *
+     * @param request Registered authenticator list request.
+     * @return Registered authenticator list response.
+     * @throws Exception In case REST API call fails.
+     */
+    RegisteredAuthenticatorsResponse getRegisteredAuthenticatorList(RegisteredAuthenticatorsRequest request) throws Exception;
```

Comment:
Isn't `Exception` too broad?

Reply:
True, I will switch to PowerAuthClientException.